insuree/tests/test_insuree_photo.py

```diff
@@ -0,0 +1,175 @@
+    test_photo_path, test_photo_uuid = "some/file/path", str(uuid.uuid4())
+
+    class BaseTestContext:
+        def __init__(self, user):
+            self.user = user
+
+    def setUp(self):
+        super(InsureePhotoTest, self).setUp()
+        self._TEST_USER = self._get_or_create_user_api()
+        self.insuree = create_test_insuree(custom_props={'chf_id': '110707070'})
+
+    @classmethod
+    def setUpClass(cls):
+        # Signals are not automatically bound in unit tests
+        super(InsureePhotoTest, cls).setUpClass()
+        cls.insuree_client = Client(Schema(mutation=insuree_schema.Mutation, query=insuree_schema.Query))
+        insuree_schema.bind_signals()
+
+    def test_add_photo_save_db(self):
+        self.__call_photo_mutation()
+        self.assertEqual(self.insuree.photo.photo, self.photo_base64)
+
+    def test_pull_photo_db(self):
+        self.__call_photo_mutation()
+        query_result = self.__call_photo_query()
+        try:
+            gql_photo = query_result['data']['insurees']['edges'][0]['node']['photo']
+            self.assertEqual(gql_photo['photo'], self.photo_base64)
+        except Exception as e:
+            print(query_result)
+            raise e
+
+    @mock.patch('insuree.services.InsureeConfig')
+    @mock.patch('insuree.services.create_file')
+    def test_add_photo_save_files(self, create_file, insuree_config):
+        create_file.return_value = self.test_photo_path, self.test_photo_uuid
+        insuree_config.insuree_photos_root_path = PropertyMock(return_value="insuree_root_path")
+
+        self.__call_photo_mutation()
+
+        self.assertEqual(self.insuree.photo.folder, "some/file/path")
+        self.assertEqual(self.insuree.photo.filename, str(self.test_photo_uuid))
+        create_file.assert_called_once_with(ANY, self.insuree.id, self.photo_base64)
+
+    @mock.patch('insuree.services.InsureeConfig')
+    @mock.patch('insuree.gql_queries.InsureeConfig')
+    @mock.patch('insuree.services.create_file')
+    @mock.patch('insuree.gql_queries.load_photo_file')
+    def test_pull_photo_file_path(self, load_photo_file, create_file, insuree_config, insuree_config2):
+        load_photo_file.return_value = self.photo_base64
+        create_file.return_value = self.test_photo_path, self.test_photo_uuid
+        insuree_config.insuree_photos_root_path = PropertyMock(return_value="insuree_root_path")
+        insuree_config2.insuree_photos_root_path = PropertyMock(return_value="insuree_root_path")
+
+        self.__call_photo_mutation()
+        query_result = self.__call_photo_query()
+        gql_photo = query_result['data']['insurees']['edges'][0]['node']['photo']
```

Comment:
@dborowiecki I think you should be careful with proceeding with indexes in such way. Could you maybe fix this?
Please look on CI too, here you have an error with 'indexes', that is why I wrote this comment. 
Maybe could be better to have check if this index exist and if yes -> assign value into this variable otherwise return null etc. 
Anyway you have the error on that section on CI and we need to check why. Maybe it is a matter of configuration on CI?

Reply:
That's a good point. In case of regular code performing logic this level of coupling would be quite unacceptable. 
However in case of  this unit test it's exactly the payload that we expect to receive for given setup. It's independent from tested part and validation or assertion is not required (bud most definitely could done in scope of separate unit test class). 

Issue with CI was related to test setup and was resolved.

Reply:
@sniedzielski  is this comment resolved  ?